apps/src/lab2/hooks/useLifecycleNotifier.ts

```diff
@@ -0,0 +1,23 @@
+import {useEffect} from 'react';
```

Comment:
@sanchitmalhotra126 tagged you on this PR for thoughts/fyi that I'm adding this hook.

Reply:
Nice, this seems really handy!